pkg/url/url.go

```diff
@@ -191,11 +283,40 @@ func List(client *occlient.Client, localConfig *config.LocalConfigInfo, componen
 	return urlList, nil
 }
 
-func getProtocol(route routev1.Route) string {
-	if route.Spec.TLS != nil {
-		return "https"
+// ListIngress returns all Ingress URLs for given component.
+func ListIngress(client *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInfo, componentName string) (iextensionsv1.IngressList, error) {
+
+	labelSelector := fmt.Sprintf("%v=%v", componentlabels.ComponentLabel, componentName)
+	// TODO: select url name
+	ingresses, err := client.ListIngresses(labelSelector)
+	if err != nil {
+		return iextensionsv1.IngressList{}, errors.Wrap(err, "unable to list ingress names")
+	}
+
+	var urls []iextensionsv1.Ingress
+
+	for _, i := range ingresses {
+		clusterURL := getMachineReadableFormatIngress(i)
+		urls = append(urls, clusterURL)
+	}
+
+	urlList := getMachineReadableFormatForIngressList(urls)
+	return urlList, nil
+}
+
+// GetProtocol returns the protocol string
+func GetProtocol(route routev1.Route, ingress iextensionsv1.Ingress) string {
+	if experimental.IsExperimentalModeEnabled() {
+		if ingress.Spec.TLS != nil {
+			return "https"
+		}
+		return "http"
```

Comment:
since "http" is returned in the else block too, put it at the end of the function

Reply:
updated.